src/metabase/api/card.clj

```diff
@@ -930,4 +933,48 @@ saved later when it is ready."
       (persisted-info/mark-for-pruning! {:id (:id persisted-info)} "off")
       api/generic-204-no-content)))
 
+(s/defn param-values
+  "Fetch values for a parameter.
+
+  The source of values could be:
+  - static-list: user defined values list
+  - card: values is result of running a card"
+  ([card param-key]
+   (param-values card param-key nil))
+
+  ([card      :- su/Map
+    param-key :- su/NonBlankString
+    query     :- (s/maybe su/NonBlankString)]
+   (let [param       (get (m/index-by :id (:parameters card)) param-key)
+         source-type (:values_source_type param)]
+     (when-not param
+       (throw (ex-info (tru "Card does not have a parameter with the ID {0}" (pr-str param-key))
+                       {:status-code 400})))
+     (case source-type
+       "static-list" (params.static-values/param->values param query)
+       "card"        (params.card-values/param->values param query)
+       (throw (ex-info (tru "Invalid values-source-type: {0}" (pr-str source-type))
+                       {:values-source-type source-type
+                        :status-code        400}))))))
```

Comment:
I wonder if the whole `case` should be extracted into a common ns. If we added another source-type it'd be too bad to have to update the dispatcher in both the card and dashboard APIs. Unfortunately with all the card-specific error messaging we can't get too DRY, but I think extracting the main branching is a good result.

Reply:
I thought about it too, but the card does not support chain-filter and it makes things awkward.

Another idea is use defmethod that dispatch with `parameterized-object-type, values-source-type`.

but I think it's a bit too much for now. Definitely need to revisit this idea once we support custom-list on data apps.